### PR TITLE
use \OCP\IUserManager::getDisplayName

### PR DESCRIPTION
**Pull request by hamza221**
ref https://github.com/nextcloud/groupware/issues/53

### CODE REVIEW

**Review by tcitworld — APPROVED** [review]

File: tests/php/unit/Service/Appointments/MailServiceTest.php [hunk 1]
```diff
@@ -422,9 +431,11 @@ public function testSendBookingInformationEmailFailed(): void {
 		$this->userManager->expects(self::once())
 			->method('get')
 			->willReturn($this->createConfiguredMock(IUser::class, [
-				'getEmailAddress' => 'test@test.com',
-				'getDisplayName' => 'Test Test'
+				'getEmailAddress' => 'test@test.com'
 			]));
+		$this->userManager->expects(self::exactly(3))
+		->method('getDisplayName')
```
Comment (ChristophWurst) [inline tests/php/unit/Service/Appointments/MailServiceTest.php:437]: indentation?

File: tests/php/unit/Service/Appointments/MailServiceTest.php [hunk 2]
```diff
@@ -349,9 +356,11 @@ public function testSendBookingInformationEmail(): void {
...
```
Comment (ChristophWurst) [inline tests/php/unit/Service/Appointments/MailServiceTest.php:362]: same

**Review by ChristophWurst — APPROVED** [review]
Very nice

File: lib/Service/Appointments/BookingCalendarWriter.php [hunk 3]
```diff
@@ -103,8 +103,8 @@ public function write(AppointmentConfig $config,
 		if (!($calendar instanceof ICreateFromString)) {
 			throw new RuntimeException('Could not find a public writable calendar for this principal');
 		}
-
-		$organizer = $this->userManager->get($config->getUserId());
+		$organizerId = $config->getUserId();
+		$organizer = $this->userManager->get($organizerId);
```
Comment (st3iny) [inline lib/Service/Appointments/BookingCalendarWriter.php:107]: Refactoring here doesn't make sense as the user object is required anyway. It should only be refactored when fetching the whole user object just to call `getDisplayName()` and nothing else.

File: lib/Service/Appointments/MailService.php [hunk 4]
```diff
@@ -90,7 +90,9 @@ public function __construct(IMailer $mailer,
 	 * @throws ServiceException
 	 */
 	public function sendConfirmationEmail(Booking $booking, AppointmentConfig $config): void {
-		$user = $this->userManager->get($config->getUserId());
+
+		$userId = $config->getUserId();
+		$user = $this->userManager->get($userId);
```
Comment (st3iny) [inline lib/Service/Appointments/MailService.php:95]: Same as above applies here.

File: lib/Service/Appointments/MailService.php [hunk 5]
```diff
@@ -272,8 +275,10 @@ private function getSysEmail(): string {
 	}
 
 	public function sendOrganizerBookingInformationEmail(Booking $booking, AppointmentConfig $config, string $calendar) {
+		
+		$userId = $config->getUserId();
 		/** @var IUser $user */
-		$user = $this->userManager->get($config->getUserId());
+		$user = $this->userManager->get($userId);
```
Comment (st3iny) [inline lib/Service/Appointments/MailService.php:281]: Same as above.

We can probably get rid of the user object completely though. It only seems to be used to check if the user actually exists some lines further down.

File: lib/Service/Appointments/MailService.php [hunk 6]
```diff
@@ -163,7 +165,8 @@ public function sendConfirmationEmail(Booking $booking, AppointmentConfig $confi
 	 * @throws ServiceException
 	 */
 	public function sendBookingInformationEmail(Booking $booking, AppointmentConfig $config, string $calendar): void {
-		$user = $this->userManager->get($config->getUserId());
+		$userId = $config->getUserId();
+		$user = $this->userManager->get($userId);
```
Comment (st3iny) [inline lib/Service/Appointments/MailService.php:169]: Same as above. Potentially could get rid of the user object here too.

**Review by miaulalala — APPROVED** [review]

**Review by st3iny — COMMENTED** [review]
A bit late but ... :see_no_evil:

**Comment by ChristophWurst** [comment]
@hamza221 follow-up with the coding style fixes and revert of unnecessary optimizations?

**Pull request merged**